tools/fuzzer/ast.h

```diff
@@ -25,6 +25,120 @@
 
 namespace fuzzer {
 
+enum class ScalarType : unsigned char;
+class TaggedType;
+class PointerType;
+class ReferenceType;
+
+using QualifiableType = std::variant<ScalarType, TaggedType, PointerType>;
```

Comment:
Should `ReferenceType` also be qualifiable?

Reply:
No, `int& *` is not a valid C++ type, hence the differentiation between `Type` (allows refs) and `QualifiableType` (doesn't allow refs).